### DISCUSSION TITLE
Workbench Beta Phase 2022-11-14 to 2023-02-20

### DISCUSSION THREAD

**Discussion by zkamvar** [LibraryCarpentry/lc-shell#211] [Announcements]
Hello @LibraryCarpentry/lc-shell-maintainers,

Thank you for your patience during the last quarter. 
I am pleased to announce that the **transition to beta phase for this lesson will take place in 6 weeks on 2022-11-14**.

## Timing

the beta phase is divided into the following schedule of stages

 - 2022-11-14: pre-beta stage (two repos, two sites)
 - 2022-12-12: beta stage (one repo, two sites)
 - 2023-02-20: pre-release stage (one repo, one site) 

Please let me know if this does not work for you and I will do my best to accommodate. 
Importantly: **please resolve (close or merge) any open pull requests before 2022-12-12** because the beta stage will involve changing the folder structure of this repository which will render any existing pull requests invalid.

## Pre Beta Tasks

During this first pre-beta stage of the Beta Phase, I will give you access to https://github.com/fishtree-attempt/{{lesson}} and encourage you to test out the following things on the lesson to get familiar with The Workbench. Here are some suggestions:

 - clone the lesson and build it locally
 - (optionally trigger and) merge and automated workflow update PRs that @carpentries-bot opens
 - review a PR that includes some failing accessibility checks
 - (R lessons only) review a PR using the preview of rendered changes
 - fix a simple typo in a lesson - perhaps in a relatively non-obvious location e.g. in the setup instructions
 - write a link from one episode to another in the same lesson 
 
I will be recording a short video describing the changes you will see in the next week, so be sure to keep your eyes out for that.

**Comment by zkamvar** [comment]
Hello @LibraryCarpentry/lc-shell-maintainers, 

Just another reminder, we will be entering this lesson into the [Workbench Beta Phase](https://carpentries.github.io/beta-phase.html) in about three weeks. 

In order to enter this phase, I require the consent of all the maintainers, so please reply with your consent (or non-consent if it may be).

**Reply by jmjamison** [reply]
Consent.   I'm behind with everything but I have this setup now in my local and department repository

**Reply by annaoates** [reply]
Thank you for the follow up. I consent as well.

**Reply by kaitlinnewson** [reply]
I consent as well.

**Comment by zkamvar** [comment]
Hello @LibraryCarpentry/lc-shell-maintainers,

This is just a heads-up that your lesson will enter the Workbench Beta Phase in just over one week.

Before that happens, #212 _must_ be merged. There is a small nuance to this lesson that is preventing me from performing [the automated conversion](https://github.com/carpentries/lesson-transition#readme) which will be fixed in #212. 

Once that PR is merged, I will create another pull request that [defines the beta phase in _config.yml](https://github.com/datacarpentry/r-raster-vector-geospatial/pull/380/files), which should be merged on 2022-11-14. 

Please let me know if you have any questions!

**Reply by zkamvar** [reply]
Hello @LibraryCarpentry/lc-shell-maintainers,

This is another reminder about the entry into the Workbench beta phase and the merging of #212. 
Unless I hear back by Thursday, I will put this lesson on hold for The Beta Phase of the workbench.

**Comment by zkamvar** [comment]
## Workbench Beta Phase Timeframe Adjustment

> Note: this is identical to the message I sent directly via email

The beta phase has been in progress since the end of October, but **because of various changing circumstances, I will be pushing back the beta phase timeline according to the revised schedule below**, synchronising all of the beta phase repositories. With this new timeline, you will have more time to explore the repository snapshots, I will have time to provide more clear communications, and—because everyone will go through each stage synchronously—you will all be able to exchange tips and tricks together. 

Note that Toby Hodges, Vini Salazar, and I will be hosting weekly helpdesk sessions, whose times we will send to you soon. 

Here is the revised schedule, which you can also find [on The Workbench beta phase website](https://carpentries.github.io/workbench/beta-phase.html):

 - [2022-12-12 (pre-beta)](https://carpentries.github.io/workbench/beta-phase.html#pre-beta): (re-)release all beta phase lessons to the pre-beta stage, which will have a parallel snapshot at https://github.com/fishtree-attempt/ hosting the Workbench preview on https://preview.carpentries.org/. 
    - During this time, we encourage you to explore the repository. Here is a [list of tasks to get started](https://carpentries.github.io/workbench/beta-phase.html#pre-beta-tasks). Remember, This preview version is temporary, so you do not have to worry about making any mistakes here. 
    - Take the time to merge or resolve any open PRs. And yes, this is a big task and it is helpful to [remember how to say 'no'](https://opensource.guide/best-practices/#learning-to-say-no) in a way that respects the contributor’s time and effort. If you feel like you need help with a particular PR, please contact Zhian and he will help craft a response.
 - [2023-02-06 (beta)](https://carpentries.github.io/workbench/beta-phase.html#beta): do in-place conversion of all beta phase lessons to use The Workbench by default. The default lesson site will still point to a snapshot of the jekyll version from 2023-02-06, and the preview lesson site will show the Workbench hosted on the main branch of the repository. 
    - IMPORTANT: before this happens, all maintainers MUST be aware of this change and must merge or resolve any open PRs. Remember that it’s okay to say no and/or contact Zhian for wordsmithing.
    - Zhian will create a video describing what this process is doing
    - The broader maintainer community will be alerted that they have three months to merge or resolve their open PRs at this time.
 - [2023-04-03 (pre-release)](https://carpentries.github.io/workbench/beta-phase.html#pre-release): The Workbench will become the default version of the lesson and the Jekyll version will be removed. A blog post will go out alerting everyone of the change. 


**We will make the conversion of all of our lessons to use The Workbench on 2023-05-01**


Thank you once again for agreeing to participate in the beta phase of The Carpentries Workbench. Your efforts in testing the infrastructure on your established lessons are invaluable to improving the workbench and I want to make sure that we take the time to get that right.

**Comment by zkamvar** [comment]
Hello @LibraryCarpentry/lc-shell-maintainers,

I wanted to apologise if my communication has been sporadic. We are now in the beta phase for this lesson and I want to make sure that people who visit the webpage can have access. To do that, please merge #212 (note: if you are wondering, yes, you can merge this before the others) or please let me know if you would like help, more explanation, or would like to not continue with the beta phase (knowing that this lesson will transition to the workbench in May). 

I know that #212 is a large PR, containing 3 years of commits from styles and this is the type of situation the workbench was built to avoid. For this instance, since it's not affecting the contents of your lesson, you do not have to review the contents.

## After merging #212 

I've sent you all invitations to access https://github.com/fishtree-attempt/lc-shell (from which https://preview.carpentries.org/lc-shell is built). This is a sandbox for you to explore and try things out until 2023-02-06, when we move to the second stage of the beta phase. 

As always, please let me know if you have any questions. You can contact me here, on slack, by email (at carpentries.org), or generally anywhere with the handle "zkamvar".

In service,
Zhian

**Comment by zkamvar** [comment]
Hello and happy new year! 

This is a reminder that on **06 February 2023 (just under four weeks away), your lesson repositories will be transformed to use The Carpentries Workbench.** This is the irreversible [second stage of the beta phase](https://carpentries.github.io/workbench/beta-phase.html#beta) where the Workbench version of the lesson will be live and the styles version will be frozen.

### Before 06 February 2023

1. **(MUST DO)** Merge or resolve any open PRs (If you are unsure of how to resolve some PRs that need more time, [here's a guide on how to say 'no'](https://opensource.guide/best-practices/#learning-to-say-no))
2. **(recommended)** Try out the [pre-beta tasks to get familiar with The Workbench](https://carpentries.github.io/workbench/beta-phase.html#pre-beta)

### After 06 February 2023

The lesson repository will change with a trimmed-down git history that reflects authorship. **To prevent accidental overwriting of repository content, I will temporarily change your access to the repository as ‘read-only’.** To regain ‘maintain’ access,

1. **(MUST DO)** [make a fresh clone of or update](https://carpentries.github.io/workbench/beta-phase.html#beta-tasks) your lesson on your computer.
2. **(MUST DO)** respond here with `I am ready for write access :rocket:`

### Have Questions?

If you have any questions, would like help resolving PRs, or anything at all, you can email, slack, or schedule a meeting with me: https://calendly.com/zkamvar.
